hashicorp/consul-k8s
# Backport of NET-581 - Added vault namespace in helm   into release/1.0.x

## Backport

This PR is auto-generated from #2841 to be assessed for backporting due to the inclusion of the label backport/1.0.x.


:rotating_light:
>**Warning** automatic cherry-pick of commits failed. If the first commit failed,
you will see a blank no-op commit below. If at least one commit succeeded, you
will see the cherry-picked commits up to, _not including_, the commit where
the merge conflict occurred.

The person who merged in the original PR is:
@absolutelightning
This person should manually cherry-pick the original PR into a new backport PR,
and close this one when the manual backport PR is merged in.

> merge conflict error: POST https://api.github.com/repos/hashicorp/consul-k8s/merges: 409 Merge conflict []



The below text is copied from the body of the original PR.

---

Changes proposed in this PR:
- Adds `vaultNamespace` in `secretsBackend.vault` in `values.yaml` 
- This namespace is used for Vault namespace, this introduces one more way to specify namespace other than
`"{"connect": [{ "ca_config": [{ "namespace": "value"}]}]}"` in `connectCA.additionalConfig`
- If `vaultNamespace` is present, it automatically adds annotation below to the templates.

```
    vault:
      agentAnnotations: |
        vault.hashicorp.com/namespace: vaultNamespace
```

How I've tested this PR:
a. CI
b. Updated test `TestVault_VaultNamespace`
Test Steps - 
```
1. kind create cluster --name=dc1
2. kind create cluster --name=dc2
3. cd acceptance/test/vault
4. go test ./... -p 1 -timeout 2h -failfast -no-cleanup-on-failure -debug-directory=/tmp/debug -use-kind -enable-multi-cluster -kube-contexts=kind-dc1,kind-dc2  -run ^TestVault
```
Output - 
```
asheshvidyut@absolutelightning-H2GX766V9T acceptance/tests (NET-581-Configure-Vault-namespaces-for-Connect-CA-via-Helm-Stanza) » go test ./... -p 1 -timeout 2h -failfast -no-cleanup-on-failure -debug-directory=/tmp/debug -use-kind -enable-multi-cluster -kube-contexts=kind-dc1,kind-dc2  -run ^TestVault
ok  	github.com/hashicorp/consul-k8s/acceptance/tests/api-gateway	0.492s [no tests to run]
ok  	github.com/hashicorp/consul-k8s/acceptance/tests/basic	0.547s [no tests to run]
ok  	github.com/hashicorp/consul-k8s/acceptance/tests/cli	0.529s [no tests to run]
ok  	github.com/hashicorp/consul-k8s/acceptance/tests/cloud	0.565s [no tests to run]
ok  	github.com/hashicorp/consul-k8s/acceptance/tests/config-entries	0.524s [no tests to run]
ok  	github.com/hashicorp/consul-k8s/acceptance/tests/connect	0.556s [no tests to run]
ok  	github.com/hashicorp/consul-k8s/acceptance/tests/consul-dns	0.543s [no tests to run]
ok  	github.com/hashicorp/consul-k8s/acceptance/tests/example	0.540s
ok  	github.com/hashicorp/consul-k8s/acceptance/tests/ingress-gateway	0.553s [no tests to run]
ok  	github.com/hashicorp/consul-k8s/acceptance/tests/metrics	0.555s [no tests to run]
ok  	github.com/hashicorp/consul-k8s/acceptance/tests/partitions	0.561s [no tests to run]
ok  	github.com/hashicorp/consul-k8s/acceptance/tests/peering	0.568s [no tests to run]
ok  	github.com/hashicorp/consul-k8s/acceptance/tests/sameness	0.552s
ok  	github.com/hashicorp/consul-k8s/acceptance/tests/snapshot-agent	0.572s [no tests to run]
ok  	github.com/hashicorp/consul-k8s/acceptance/tests/sync	0.554s [no tests to run]
ok  	github.com/hashicorp/consul-k8s/acceptance/tests/terminating-gateway	0.553s [no tests to run]
ok  	github.com/hashicorp/consul-k8s/acceptance/tests/vault	916.339s
ok  	github.com/hashicorp/consul-k8s/acceptance/tests/wan-federation	0.587s [no tests to run]
```

How I expect reviewers to test this PR:


Checklist:




---

<details>
<summary> Overview of commits </summary>

  - bc79b0a235af21a9747e05a05c8655748be7ee42  - 3ee878c6a3867193b9559b27ca22a514d14f51ef  - 10ec942ff1902a73222d47b09695d3836d79af68  - 9935a1b6fda54895e90d05386829b4cc279a8f1b  - 4097b48e6757d3c2c8e9818c7e5ec2b0d286527a  - 3f8c4f3e66efa57cabf91aa32604cc941d31b0b4  - 5339dd9e5238a37951552613349d73d42246d940  - c7342840f1e1f0ca6ec613332ae3dfdd12be19cc  - 9e2f11deb3412657831c1dece906bfe866447150  - 9d2ba12f6c6d51d7ddba7b68ec96d2e40f704c52  - 8cc2117b8c5f9480219292871eb36d561a3dffec  - 01f3e9568d98bd79fda9fc28eb07989a36115906  - 6917abb6b06b7efbe7cc4b9b4b1eadfa652f0a5a  - ad710bf6736a2ab257899aafae867043c15e33d1  - fdd7cf1393dee436dad7cca8e30a69cd7354fe45  - d3e0323dbb82bb102d039e30feae3646b77a1864  - 7b3482bc7b19339085164eb997b1c8e2d706ac53  - 532210d4f669f3fc472e91ec9ea690303dffce87  - fd469b6a8dbb1865d0ca0ab829ace7f10616a6af  - 7c870eab686f83ac84fe9a1bef2526f483a44239  - cac8d3a58c983132caf5d22c099ba4dc7c2487ac  - c4a0f8a4a20cd7041cd00ba5f52a8a4149dddb93  - 6c692df4272de49824830b07435a073565f120e5  - d05f34092b132391fccaa4d51e464d495e92dd72  - 261f3799989ab2652c7ba49dc60d42d79132de63  - 3d9c05cf4c05d40e6de839aa8ffd85c4171bce4c  - 228b423a34da8ef91a8f00cb8f21e3faffd1a542  - ef1710de054a763169c8b487097d3b5c170cc249  - 0827f81b9687c033459a1e3e5141d9afc1a535ef  - 306632b9b6da7d467690ea1db3dbeeb40433c6c2  - faec4c2130955d3927eeaf64103c8e668f3bc362  - b1fd0fded2d2381c559c91139677853c594f435e  - 636208f03bd7f3052ffda93838c24371ff26996f  - 666331a27729ecf628cd300b29afb2de64349ef1  - ffe4b1fda28ff42b917dd25e71f45b36a04346a1  - c8478a0a374848ed90535ddddd6647c865e71d95  - ecfff91b0a9dd0731f216d8d9096f977eca07560  - 959431ff8d6796152497ef64274341ff3d779bf9  - 8f52b7c98e6f5b761c0ac69798c28ae152178de1  - 1f3e23ad70ae687da8453c9b66d44b77df63f8cc  - ed1026e7b0d89aa958af1c493864895b88c3f57a  - 825fb42f0d7ed58f114b57d1dc6df2cc7e614a5b  - b840d326083c8fa3be598fc6c7e6610a6a9621a3  - ed2573afc5c98ee3b5c716796d454ba497b3b4f2  - a47a3d850e5648680e3a33de4f8ecea068c2ae3a  - 8a7b8bd1b7950eca0bf95dcbc7d99fb0afa77a32  - 3c4291cf2f4c0b540e3d14675be8cd90115d15fa  - 4523282b37b1f750fdc3c35b61e47e85093b00c6  - af7e617c395fb0bff06f41d30624169bf04ac40a  - 124cc4d3bc7da779995fb7b96ab75ac5a5cbacdd  - 4be3cea6d4365f60b459f2e6b9db2a0081d09e48  - 7c13074652658614753b95e0af0550a3cd706691  - d6da52ebd30033ecd5681a647375b6d0140ac5e7  - 5f6a20077741b9f32898d6f7cb77147e344bdd7d 

</details>
